services-core/src/gas_price.rs

```diff
@@ -23,14 +25,48 @@ pub trait GasPriceEstimating: Send + Sync {
     async fn estimate_with_limits(&self, gas_limit: f64, time_limit: Duration) -> Result<f64>;
 }
 
+#[async_trait::async_trait]
+pub trait Transport: Send + Sync {
+    async fn get_json<T: DeserializeOwned>(&self, url: &'static str) -> Result<T>;
```

Comment:
Not sure how to best pass the URL. I didn't want to commit to some specific URL library.

Reply:
I think you just need to specify a lifetime parameter:
```suggestion
    async fn get_json<'a, T: DeserializeOwned>(&self, url: &'a str) -> Result<T>;
```

There might be issues with the `async_trait` (as the boxed future result will have its lifetime tied to this string), maybe @e00E knows more as I'm not super familiar with it.

Reply:
Should be `&str` without `'static`. It is true that for our current clients all the urls are static but in general they don't have to be.

Reply:
Passing it as str instead of a specific Url type is fine imo. All urls can be converted to/from strings so the only the disadvantage is a small performance loss because we might be constructing an url twice.

Reply:
> I didn't want to commit to some specific URL library.

You could always use [`http`](https://docs.rs/http/0.2.1/http/) which is what many HTTP clients in Rust use (like hyper, isahc, etc.). I'm also 100% fine with using a string here as well.